src/data_processing/wiki_to_netflix.py

```diff
@@ -0,0 +1,136 @@
+import requests
+import csv
+
+base_dir = os.path.join(os.path.dirname(__file__), '../data/')
+user_agent = 'Noisebridge MovieBot 0.0.1/Audiodude <audiodude@gmail.com>'
+
+missing_count = 0
+num_rows = 42
+
+# Reading netflix text file
+def read_netflix_txt(txt_file):
+    with open(txt_file, "r", encoding = "ISO-8859-1") as netflix_data:
+        netflix_list = [line.rstrip().split(',', 2) for line in netflix_data.readlines()[0:num_rows]]
+    return(netflix_list)
+
+# Writing netflix csv file
+def create_netflix_csv(csv_name, data_list):   
+    with open(csv_name, 'w') as netflix_csv:
+        csv.writer(netflix_csv).writerows(data_list)
+
+# Extracting movie info from Wiki data
+def wiki_feature_info(data, key):
+    if len(data['results']['bindings']) < 1 or key not in data['results']['bindings'][0]:
+        return 'NA'
```

Comment:
Is there a reason you're using `'NA'` and not just `None`?

Reply:
It is kind of a holdover from the previous method of reading and writing to the same csv and adding NA to the missing columns. Which was refactored to creating the csv and adding only the found rows at the end.

Reply:
Do you think we should change it to `None`?

Reply:
I changed it to "NULL" so that Mongo stores unmatched movies with a wikidata_id (and other feature info) of NULL (based on the comment in the other issue); Curious if/why None is typically preferred?

Reply:
For Mongo, it's okay to simply leave out fields that aren't known or required. So you can have one movie:

```
{
  "title": "Jurassic Park",
  "year": "1996",  # Note, this is when it actually makes sense to have the year as an int!
  "wikidata_id": "Q56789",
  "genres": ["Adventure", "Family", "Action"]
  "director": "Stephen Spielberg"
}
```

and another movie:

```
{
  "title": "Wacky Adventures With Fruits",
  "year": "1999",
}
```

And they can both exist in the same Mongo collection. When reading the data, we will have to do a check like `if 'wikidata_id' in movie_data:`, but this is no more work then doing `if movie['wikidata_id'] == 'NULL'` and the database is cleaner.

Reply:
There is a big difference between the string `"NULL"` and the value `None`.

The first is a string, obviously. It doesn't have any special meaning. It might as well be "APPLE".

`None` signifies the absence of a value. Think of fruits. If we don't know the price of a fruit, but we know we have the fruit, we can use `None` to signify that the value is unknown/not present:

```
fruits = {'apple": 1.29, 'banana': None, 'kiwi": 1.89}
```

We need something to put after the banana key, but we can't put 0 (that would imply that bananas are free). We could maybe use what's called a "sentinel value" of -1 or something, but then someone who is calculating the price of all the fruits might not realize that the sentinel is there:

```
total_price = sum(fruits.values())
```

On the other hand, if the value is `None`, the above code will raise an exception. I sometimes find it easiest to think of numbers/integers. If the bank balances is 1, you have one dollar. If it's -1, you owe one dollar. If it's 0, you have no money. How do we specify that your account is closed and there is "no" balance? We use `None`.

The other thing that may be confusing is that in other languages like SQL, the name `NULL` is used (no quotes!) for the absence of a value such as:

```
...WHERE wikidata_id IS NULL
```

In Ruby, this is called `nil`. It's all the same thing: there is no value (very zen)